internal/jobcontainers/path.go

```diff
@@ -73,8 +72,10 @@ func getApplicationName(commandLine, workingDirectory, pathEnv string) (string,
 		result     string
 	)
 
-	// Clean the path, to get rid of any . elements
-	commandLine = filepath.Clean(commandLine)
+	// Handle a relative path with .
+	if len(commandLine) >= 3 && commandLine[0:2] == "./" || commandLine[0:2] == ".\\" {
```

Comment:
Note that this will not work in a case like `.\.\foo.exe`. Is there a more robust way to handle this that doesn't require us writing path parsing code?

Reply:
Was wondering the same, I don't remember cexecsvc doing anything with this so I'm wondering how it worked there (if it even does)

Reply:
Well, I guess we could call `filepath.Clean` on every search path we're currently trying. The issues only really for arguments that we don't want to be converted to forward slashes, but if we got as far as the arguments and haven't found an exe, then we're doomed to fail anyways.

Reply:
Ok after testing this, at least just through the crictl -> ctrd -> shim flow (not sure if docker used to do any cmdline/arg touchups), I don't think relative paths (with a dot) work for Windows Server containers either. I'm setting my working directory to C:\\ and have a binary named test.exe at the root that's failing to be found with a command line of "./test.exe" or ".\\test.exe". It's successfully found with just "loop.exe". 

Putting test.exe in a subdirectory foo, "foo\\test.exe" and "foo/test.exe" work, but ".\\foo\\test.exe" and "./foo/test.exe" don't.

Reply:
If our cmdline parsing is expected to stay fairly similar to Windows Server containers, we might be a bit ahead in this regard.. unless I'm missing something which is always possible 😆

Reply:
I see. Is it required that we support a relative path for job containers?

Reply:
I don't know if it's "required" really, but would be nice. A user can supply a relative path without dot and get by just like for server containers. Mainly looked into this as @claudiubelu had ran into an issue supplying a dotted relative path and informed me, @jsturtevant as well.